libs/entitlement/qa/entitlement-test-plugin/src/main/java/org/elasticsearch/entitlement/qa/test/DummyImplementations.java

```diff
@@ -676,4 +690,162 @@ public void setAttribute(Path path, String attribute, Object value, LinkOption..
 
         }
     }
+
+    static class DummyFileChannel extends FileChannel {
```

Comment:
Wondering, that's a lot of code to just test `close` and it's not directly obvious what this is used for. Maybe add a comment or is it simpler in the end to create a channel via the entitled plugin?

Reply:
Wait, makes sense, I misread the test... we reject subclassing 👍